packages/modal-ui/src/views/wcm-mobile-connecting-view/index.ts

```diff
@@ -26,23 +26,25 @@ export class WcmMobileConnectingView extends LitElement {
     const { mobile, name } = CoreUtil.getWalletRouterData()
     const nativeUrl = mobile?.native
     const universalUrl = mobile?.universal
-
+    const target = CoreUtil.isTelegram() ? '_blank' : '_self'
+    // eslint-disable-next-line no-param-reassign
+    uri = CoreUtil.isTelegram() && CoreUtil.isAndroid() ? encodeURIComponent(uri) : uri
     if (nativeUrl && !forceUniversalUrl) {
       const href = CoreUtil.formatNativeUrl(nativeUrl, uri, name)
-      CoreUtil.openHref(href, '_self')
+      CoreUtil.openHref(href, target)
```

Comment:
Same here, seems they are getting the same params, we could call it once, outside of the if/else clauses?